src/Tribe/REST/V1/Endpoints/Single_Event.php

```diff
@@ -479,4 +431,141 @@ public function can_delete() {
 
 		return current_user_can( $cap );
 	}
+
+	/**
+	 * Handles UPDATE requests on the endpoint.
+	 *
+	 * @param WP_REST_Request $request
+	 * @param bool            $return_id Whether the created post ID should be returned or the full response object.
+	 *
+	 * @return WP_Error|WP_REST_Response An array containing the data of the updated post on
+	 *                                   success or a WP_Error instance on failure.
+	 */
+	public function update( WP_REST_Request $request ) {
+		$this->serving = $request;
+
+		$postarr = $this->prepare_postarr( $request );
+
+		if ( is_wp_error( $postarr ) ) {
+			return $postarr;
+		}
+
+		$id = Tribe__Events__API::updateEvent( $request['id'], array_filter( $postarr ) );
+
+		if ( is_wp_error( $id ) ) {
+			/** @var WP_Error $id */
+			return $id;
+		}
+
+		if ( false === $id ) {
+			$message = $this->messages->get_message( 'could-not-update-event' );
+
+			return new WP_Error( 'could-not-update-event', $message, array( 'status' => 403 ) );
+		}
+
+		$data = $this->post_repository->get_event_data( $id );
+
+		$response = new WP_REST_Response( $data );
+		$response->set_status( 200 );
+
+		return $response;
+	}
+
+	/**
+	 * Returns the content of the `args` array that should be used to register the endpoint
+	 * with the `register_rest_route` function.
+	 *
+	 * @return array
+	 */
+	public function EDIT_args() {
+		// when editing the only required argument is the ID ('id')
+		$create_args = $this->CREATE_args();
+		array_walk( $create_args, array( $this, 'unrequire_arg' ) );
+
+		return array_merge( $this->READ_args(), $create_args );
+	}
+
+	/**
+	 * @return bool Whether the current user can update or not.
+	 */
+	public function can_update() {
+		return $this->can_post();
+	}
+
+	/**
+	 * Prepares an array with the event data from the specified request.
+	 *
+	 * @param WP_REST_Request $request
+	 *
+	 * @return array|WP_Error An array specifying the event data, a `WP_Error` if an error occurred.
+	 */
+	protected function prepare_postarr( WP_REST_Request $request ) {
+		$post_object           = get_post_type_object( Tribe__Events__Main::POSTTYPE );
+		$can_publish           = current_user_can( $post_object->cap->publish_posts );
+		$can_edit_others_posts = current_user_can( $post_object->cap->edit_others_posts );
+		$events_cat            = Tribe__Events__Main::TAXONOMY;
+
+		$post_data     = isset( $request['date'] ) ? Tribe__Date_Utils::reformat( $request['date'],
```

Comment:
Seems just a _little_ funky to break everything between `?` and `:` over two lines. Here and other places we're doing this, could we stick to something like:

```php
$data = isset( $a )
	? foo( 123 )
	: $bar;
```